### ISSUE TITLE
Milestone: release tooling for foolproof hybrid encryption

### ISSUE THREAD

**Issue by CluEleSsUK** [drand/roadmap#35]
eta: 2023-09-30

Given the resource limitations of the FVM, it does not make sense to perform symmetric decryption on content stored in it. To that end, we must create tooling to make it seamless and bug-free for developers building applications with timelock encryption on FVM.

**Comment by nikkolasg** [comment]
I disagree: it makes a lot of sense to decrypt onchain for applications and thus we should provide an encryption mechanism that is cheaply decryptable onchain.(Stg like using keccak(1 || privKey) for the first 32bytes then keccak(2 || privKey) for the second 32bytes etc)

**Comment by CluEleSsUK** [comment]
Reasonable - we will have to design exactly how that looks; it could be that we provide 2 separate actors on filecoin and/or auxiliary tooling for those that want to do some operations off-chain
